### PR TITLE
Update the 3.13 plan

### PR DESCRIPTION
**Pull request by mdboom**
This builds on Mark's excellent overview.

1. I broke out copy-and-patch into its own thing -- I think it's big enough and parts of it (especially the build-time stuff), is parallelizable.
2. I added PEP 554 (subinterpreters in the stdlib).
3. For the detailed planning parts, I've created METABUG issues where we can link to ideas and/or cpython issues related to each step. I find issue links showing the status (open/closed etc.) of the issue extremely useful for tracking progress, but those only work issue-to-issue, not in a file or wiki.
4. For full transparency, I'm debating how much to prioritize the memory management work (at the expense of other things). I'm leaving it here for now, but I think mainly to see if there are quick cheap experiments we can run to generate more concrete solutions.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Your PR just deletes a file, maybe you forgot to git add some new files?

**Comment by brandtbucher** [comment]
Hm. I just see a deleted file. `git add -a`?

**Comment by mdboom** [comment]
Doh.  Didn't commit the meat of the work ;)

File: 3.13/README.md [hunk 1]
```diff
@@ -1,58 +1,87 @@
...
+Unlike other JITs, copy-and-patch combines a fairly complex build-time system to
+create templates, with a very simple run-time component to build machine code
+from those templates. We have already developed a working prototype for
+CPython's existing Tier 1 interpreter, and the next step is to get it
+production-ready for use with the Tier 2 interpreter described above.
+
+The main chunks of remaining work are:
+* Improve the build-time component to lower its maintenance cost.
+  * This includes, for example, reusing existing components for parsing object files, rather than writing our own.
+* Improve the integration of copy-and-patch into CPython's build system.
```
Comment (brandtbucher) [inline 3.13/README.md:56]:
```suggestion
* Improve the integration of copy-and-patch into CPython's build system.
* Improve the quality of the generated code.
  * There are many unimplemented ideas from the paper here, including pinning locals or stack items in registers across calls, "burning in" constants, and more.
```

**Review by brandtbucher — APPROVED** [review]
Nice, one small addition for now:

**Review by gvanrossum — COMMENTED** [review]
LG.

File: 3.13/README.md [hunk 2]
```diff
@@ -1,58 +1,89 @@
...
   * Implement the specializer
   * Implement the partial evaluator
-  * Implement the copy-and-patch machine code generator
 
 Our goal for 3.13 is to reduce the time spent in the interpreter by at least 50%.
 
-[Detailed plan][tier2_plan.md]
+[Detailed plan](https://github.com/faster-cpython/ideas/issues/587).
+
+### Copy-and-patch machine code generator
```
Comment (markshannon) [inline 3.13/README.md:42]: This belongs in the detailed plan for the tier 2 optimizer.
The machine code generator is part of the optimizer.

**Review by markshannon — COMMENTED** [review]
I want to keep to two main themes:
The tier 2 optimizer and memory management.

Could you move the copy-and-patch stuff into the tier 2 detailed plan, and the multiple interpreters part into an "other work" section.

File: 3.13/README.md [hunk 3]
```diff
@@ -1,58 +1,89 @@
...
-* Experiments on the register machine are complete
-* We have a viable apporach to create a low-overhead maintainable machine code generator
+* The interpreter generator is working.
+* Experiments on the register machine are complete.
+* We have a viable approach to create a low-overhead maintainable machine code generator, based on [copy-and-patch](https://fredrikbk.com/publications/copy-and-patch.pdf).
 
-We can work on the tier 2 optimizer chain unimpeded.
+We plan four parallelizable pieces of work for 3.13:
 
-We plan two main pieces of work for 3.13:
```
[the remark this answers is not in the document]
Comment (mdboom) [reply 3.13/README.md:19]: I agree about the need to retain focus.

- I'm fine with incorporating copy-and-patch under tier 2 interpreter. My main reason for separating it was (1) it has some multi-step complexity itself, and (2) large parts (but not all) is parallelizable with the tier 2 intepreter.  But I don't feel strongly.
- I want to keep PEP 544 explicitly called out.  While almost entirely independent of the rest, it is an important work product of this team for 3.13.
- I'm less certain we should do the memory management work at all at this time. But that's a complex enough discussion that we should probably move it online to a meeting.

File: 3.13/README.md [hunk 4]
```diff
@@ -1,58 +1,89 @@
...
 * The tier 2 optimizer
-* Faster memory allocation and deallocation, and reduced cycle GC ovehead 
+* Copy-and-patch machine code generator
+* Enabling subinterpreters from Python code ([PEP 554](https://peps.python.org/pep-0554/)).
```
[the remark this answers is not in the document]
Comment (mdboom) [reply 3.13/README.md:21]: I think this document as encompassing the work of the entire team, so I think it should be included (and it doesn't really fit with or relate to the Tier 2 interpreter).

**Pull request merged**